File: backend/src/main/java/teamx/app/backend/models/Task.java
package teamx.app.backend.models;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;

@Data
@Builder
@Entity(name = "Tasks")
@NoArgsConstructor
@AllArgsConstructor
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "task_order")
    private int taskOrder;

    private String name;
    private String description;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private Date deadline;

    @Enumerated(EnumType.STRING)
    private Status status;

    @ManyToOne
    @JsonIgnore
    private Project project;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "personaltodolistowner_id")
    private User personalTodoListOwner;

    public enum Status {
        TODO, IN_PROGRESS, DONE
    }
}

File: frontend/src/models/Task.js
import {Model} from "@/models/Model";

export class Task extends Model {
    constructor(id, name, description, deadline, status, personalTodoListOwnerId, personalTodoListOwnerName) {
        super();
        this.id = id;
        this.name = name;
        this.description = description;
        this.deadline = deadline;
        //this.priority = priority;
        this.status = status;
        this.personalTodoListOwnerId = personalTodoListOwnerId;
        this.personalTodoListOwnerName = personalTodoListOwnerName;
    }

    static priorityList = [
        "LOW",
        "MEDIUM",
        "HIGH"
    ];

    static statusList = [
        { value: "TODO", displayName: "TODO"} ,
        { value: "IN_PROGRESS", displayName: "In Progress" } ,
        { value: "DONE", displayName: "Done"}
    ];

}